pyannote/audio/tasks/joint_task/speaker_diarization_and_embedding.py

```diff
@@ -0,0 +1,1032 @@
+            metadatum = dict()
+
+            # keep track of source database and subset (train, development, or test)
+            if file["database"] not in metadata_unique_values["database"]:
+                metadata_unique_values["database"].append(file["database"])
+            metadatum["database"] = metadata_unique_values["database"].index(
+                file["database"]
+            )
+            metadatum["subset"] = Subsets.index(file["subset"])
+
+            # keep track of speaker label scope (file, database, or global) for speaker diarization protocols
+            if isinstance(self.protocol, SpeakerDiarizationProtocol):
+                metadatum["scope"] = Scopes.index(file["scope"])
+                # add the file to the embedding or diarization list according to the file database speaker
+                # labels scope
+                if file["scope"] == 'global':
+                    self.embedding_database_files.append(file_id)
+                elif file["scope"] in ["database", "file"]:
+                    self.diarization_database_files.append(file_id)
+
+            # keep track of list of classes for regular segmentation protocols
+            # Different files may be annotated using a different set of classes
+            # (e.g. one database for speech/music/noise, and another one for male/female/child)
+            if isinstance(self.protocol, SegmentationProtocol):
+
+                if "classes" in file:
+                    local_classes = file["classes"]
+                else:
+                    local_classes = file["annotation"].labels()
+
+                # if task was not initialized with a fixed list of classes,
+                # we build it as the union of all classes found in files
+                if self.classes is None:
+                    for klass in local_classes:
+                        if klass not in classes:
+                            classes.append(klass)
+                    annotated_classes.append(
+                        [classes.index(klass) for klass in local_classes]
+                    )
+
+                # if task was initialized with a fixed list of classes,
+                # we make sure that all files use a subset of these classes
+                # if they don't, we issue a warning and ignore the extra classes
+                else:
+                    extra_classes = set(local_classes) - set(self.classes)
+                    if extra_classes:
+                        warnings.warn(
+                            f"Ignoring extra classes ({', '.join(extra_classes)}) found for file {file['uri']} ({file['database']}). "
+                        )
+                    annotated_classes.append(
+                        [
+                            self.classes.index(klass)
+                            for klass in set(local_classes) & set(self.classes)
+                        ]
+                    )
```

Comment:
Can probably be removed. Double check, though.

Reply:
Indeed, this can be removed. Same for lines 419-423 I think. I can also remove `self.classes` and `self.annotated_classes`. These two attributes are not used anywhere in the code aside in `setup` for `SegmentationProtocol`

Reply:
Done in 3d295dd